pkg/registry/porch/background.go

```diff
@@ -28,18 +28,20 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func RunBackground(ctx context.Context, coreClient client.WithWatch, cache cache.Cache) {
+func RunBackground(ctx context.Context, coreClient client.WithWatch, cache cache.Cache, RepoSyncFrequency time.Duration) {
 	b := background{
-		coreClient: coreClient,
-		cache:      cache,
+		coreClient:        coreClient,
+		cache:             cache,
+		RepoSyncFrequency: RepoSyncFrequency,
```

Comment:
I understand the logic here of reusing the RepoSyncFreq but it is a bit of a blunt instrument.
The Watch mech here should be the main focus is my opinion.
If this is a somewhat temp solution we should prob leave some breadcrumb comments.